Sources/AblyAssetTrackingCore/Model/Location.swift

```diff
@@ -100,4 +106,66 @@ public struct Location: Equatable {
         self.floorLevel = nil
         self.timestamp = .zero
     }
+    
+    /**
+     Utility function that coerces NaN values to sensible defaults or negative values.
+     */
+    func sanitize() -> Location {
+        return Location(coordinate: coordinate,
+                        altitude: altitude,
+                        ellipsoidalAltitude: ellipsoidalAltitude.isFinite ? ellipsoidalAltitude : .zero,
```

Comment:
I think it's mainly stylistic and in our case doesn't really make a difference. It could be important if we were doing some operations on generic types here, or conforming to `AdditiveArithmetic` protocol.

I actually saw a convention that `.zero` should be used on non-primitive types, while `0` should be used for primitives (so we should be specifying `0`s here), but I don't think it's an approach that's generally agreed on by the community, and since we had been using `.zero`s here I didn't change this approach.

Reply:
Cool, thanks for the explain :)